### ISSUE TITLE
Multiple LDAP Server

### ISSUE THREAD

**Issue by mwolfe1** [snipe/snipe-it#6685]
Would it be possible to add multiple LDAP servers to sync? We manage technology for several school districts and would like to set up each district as a separate company. Each district has their own domain forest that has a trust back to our domain.

**Comment by stbc** [comment]
+1 multiple servers would be helpful regarding a fallback on a secondary if primary is down

**Comment by jelockwood** [comment]
I agree this sounds a useful enhancement. An alternative approach would be to have some sort of DNS load balancer so that e.g. ldap.domain.com redirects to a working LDAP server.

However apart from requiring the installation and setup of a DNS load balancer care also needs to be taken over possible issues with and SSL certificates and their subject alternative names so that when redirected from say ldap.domain.com to ldap1.domain.com the certificates still work.

**Comment by stbc** [comment]
Well, yeah, that setup is common with a lb in front of two (or more)  LDAP servers. We are running a setup like that right now and have no issues with that so far. But it still would be a nice feature to add a fallback server for people not having a quite complex load balancing scenario...

**Comment by stbc** [comment]
Yes. Please. See #7087 as well.

**Comment by stbc** [comment]
Dear stale bot: Yes!

**Comment by fryguy503** [comment]
This request is still very relevant! This could help with companies that tend to acquire/divest many sub businesses. It makes it easy to bring on a new business, then just export the inventory when the segment is divested.

**Comment by mwolfe1** [comment]
Still relevant 

Michael Wolfe

**Comment by billtzim** [comment]
Yes, this feature would be great to be implemented. Very helpful to support multiple ldap servers. Thank you for your great work!

**Comment by cliff-LINKS** [comment]
+1

**Comment by stbc** [comment]
+1 Would still love to see that

**Comment by Bakan0** [comment]
Still relevant! 😅

**Comment by mwolfe1** [comment]
Still relevant 

Michael Wolfe

**Comment by mwolfe1** [comment]
Still relevant 

Michael Wolfe

**Comment by AisFlo** [comment]
We would like this Feature too, as a fallback option.
I guess it could be a problem to implement, because with multiple LDAP server, you have to compare them to the existing Users.
What if they differ?

**Comment by jelockwood** [comment]
@AisFlo 
The use I am anticipating would be to define two (or more) connections to different members of an LDAP cluster. All cluster members should already be synchronising data so should be identical. This benefit is that if the primary defined LDAP server fails it will be able to use a secondary entry and therefore continue to provide service. As such there will be no need to worry about differences between the LDAP servers.

**Comment by fryguy503** [comment]
I am more interested in multiple separate ldap systems. E.g. I work for a company that has central management and r-wan connection to a number of schools. I would like to have companies setup for each district and users pulled from each of their AD systems for those "companies".

**Comment by bcourtade** [comment]
+1 fryguy503 
We are set up in a similar way.  Schools with separate AD Domains connected over shared fiber.  

Selecting which LDAP source to use could be as simple as a domain dropdown option on the login screen, or choosing email address as the username for matching.

**Comment by fryguy503** [comment]
Stalebot seems counter-productive, these requests are valid and should be exempted from the constant need to bump these requests.

**Comment by mwolfe1** [comment]
Still relevant. 

Michael Wolfe

**Comment by SpearRIT** [comment]
This is definitely still relevant and I believe that both the ability to add multiple ldap servers for the same domain and the ability to add multiple ldap domains are equally useful.
So... +1!

**Comment by padeli** [comment]
I'd love to see this feature rolled out!

**Comment by opayemim** [comment]
I will love to see this featured rolled out too

**Comment by OMFCP** [comment]
Ran into a similar issue trying to get my child domains to sync via ldap.  To get the users in my child domains to sync all I had to do was add the port for the ldap global catalog server connection 3268 to the end of my ldap server entry in the settings.  I looked like ldap://server.domain:3268 and it immediately pulled in all users from all of my child domains.

**Comment by derrynj** [comment]
+1 it would be very useful to have a fallback server to specify

**Comment by technomooney** [comment]
+1 to multiple LDAP connections for cross company assets.

**Comment by AntonyBragg** [comment]
+1 to multiple LDAP connections for cross company assets.

**Comment by noc-cgesd** [comment]
+1 to multiple LDAP connections. We manage assets for multiple orgs.

**Comment by kronoslostsoul** [comment]
+1 to multiple LDAP connections, please. We have a multi-domain org with different companies and it would definitely help.
